conans/test/unittests/tools/cmake/test_cmakedeps.py

```diff
@@ -143,3 +143,52 @@ def test_cmake_deps_links_flags():
             data_cmake = files["mypkg-release-x86-data.cmake"]
             assert "set(mypkg_SHARED_LINK_FLAGS_RELEASE -NODEFAULTLIB;-OTHERFLAG)" in data_cmake
             assert "set(mypkg_EXE_LINK_FLAGS_RELEASE -OPT:NOICF)" in data_cmake
+
+
+def test_component_name_same_package():
+    """
+    When the package and the component are the same the variables declared in data and linked
+    to the target have to be the same.
+    https://github.com/conan-io/conan/issues/9071"""
+    conanfile = ConanFile(Mock(), None)
+    conanfile._conan_node = Mock()
+    conanfile._conan_node.context = "host"
+    conanfile.settings = "os", "compiler", "build_type", "arch"
+    conanfile.initialize(Settings({"os": ["Windows"],
+                                   "compiler": ["gcc"],
+                                   "build_type": ["Release"],
+                                   "arch": ["x86"]}), EnvValues())
+    conanfile.settings.build_type = "Release"
+    conanfile.settings.arch = "x86"
+
+    cpp_info = CppInfo("mypkg", "dummy_root_folder1")
+
+    # We adjust the component with the same name as the package on purpose
+    cpp_info.components["mypkg"].includedirs = ["includedirs1"]
+
+    conanfile_dep = ConanFile(Mock(), None)
+    conanfile_dep.cpp_info = cpp_info
+    conanfile_dep._conan_node = Mock()
+    conanfile_dep._conan_node.context = "host"
+
+    with mock.patch('conans.ConanFile.ref', new_callable=mock.PropertyMock) as mock_ref:
```

Comment:
This patch can be avoided, assigning the ref above directly (same as done with "host" context, like in https://github.com/conan-io/conan/pull/9062/files, this allows to simplify the test a bit.

Not a big deal, can be merged, in any case it will conflict with https://github.com/conan-io/conan/pull/9062, so I can update that PR later.